lib/remote-objects.js

```diff
@@ -707,22 +708,90 @@ RemoteObjects.prototype.getScope = function(ctx, method) {
  * Define a named type conversion. The conversion is used when a
  * `SharedMethod` argument defines a type with the given `name`.
  *
- * ```js
- * remotes.defineType('MyType', function(val, ctx) {
- *   // use the val and ctx objects to return the concrete value
- *   return new MyType(val);
+ * ```
+ * remotes.defineType('MyType', {
+ *   fromTypedValue: function(ctx, value) {
+ *     // use the val and ctx objects to return the concrete value
+ *     if (value === undefined || value === null)
+ *       return { value: value };
+ *
+ *     value = new MyType(value);
+ *     var error = this.validate(ctx, value);
+ *     return error ? { error: error } : { value: value };
+ *   },
+ *
+ *   fromSloppyValue: function(ctx, value) {
```

Comment:
Good point. I took the concept of "sloppy" values from #264 but on the second thought the name is not descriptive enough.

The sloppy conversion/coercion is invoked when the value comes from a string-only source like HTTP header or more importantly a query string parsed via `qs`. 

A sloppy value is one of:
 - a string
 - an array containing sloppy values
 - an object where property values are sloppy

I am tempted to rename `fromSloppyValue` to `fromStringValue`, but that would hide the fact that an array or an object can be passed in too.

@ritch @richardpringle Any suggestions for a better name?

I reworked the comments in 0b7df4fa2 to make this matter more clear, PTAL.